Directory.Build.props

```diff
@@ -1,17 +1,30 @@
 <Project>
 
-  <Import Project="FSharpBuild.Directory.Build.props" Condition = " '$(FSharpTestCompilerVersion)' == '' "/>
-  <Import Project="FSharpTests.Directory.Build.props" Condition = " '$(FSharpTestCompilerVersion)' != '' "/>
-
-  <!-- 
+  <PropertyGroup>
+    <RepoRoot Condition="'$(RepoRoot)' == ''">$(MSBuildThisFileDirectory)</RepoRoot>
+  </PropertyGroup>
+  <!--
    When developers load the FSharp.Compiler.Service solution we set FSHARPCORE_USE_PACKAGE to true if it hasn't already been set to a value.
    This option ensures that building and testing uses the specified FSharp.Core nuget package instead of the local
    FSharp.Core project.
-   -->
-  <PropertyGroup Condition="'$(FSHARPCORE_USE_PACKAGE)' == ''">
-    <FSHARPCORE_USE_PACKAGE Condition="'$(SolutionName)' == 'FSharp.Compiler.Service'">true</FSHARPCORE_USE_PACKAGE>
+   We also disable arcade and reset certain artifacts and compiler paths to use default ones
+   All settings below can be overriden via CLI switches if needed. -->
+  <PropertyGroup Condition="'$(SolutionName)' == 'FSharp.Compiler.Service' and '$(BUILDING_FCS_SOLUTION)' != 'false'">
+    <DisableAutoSetFscCompilerPath>false</DisableAutoSetFscCompilerPath>
+    <BUILDING_FCS_SOLUTION Condition="'$(BUILDING_FCS_SOLUTION)' == ''">true</BUILDING_FCS_SOLUTION>
+    <FSHARPCORE_USE_PACKAGE Condition="'$(FSHARPCORE_USE_PACKAGE)' == ''">true</FSHARPCORE_USE_PACKAGE>
+    <DISABLE_ARCADE Condition="'$(DISABLE_ARCADE)' == ''">true</DISABLE_ARCADE>
+    <ArtifactsDir Condition="'$(ArtifactsDir)' == ''">$(MSBuildThisFileDirectory)artifacts/</ArtifactsDir>
```

Comment:
I wonder if reusing the same artifacts directory may make `obj` and `bin` directories be reused in an unexpected way. For example, there're files that are generated during NuGet restore, and dependencies would be different between FSharp.sln and FSharp.Compiler.Service.sln. I'm also not sure if a solution build would rebuild things correctly if switching the solutions or having both open at the same time.

Will having a separate artifacts directory an easy change here? Something like `artifacts/fcs`, for example?

Reply:
It's not reusing same artifacts dir now (i.e this change doesn't change it).
It will have to be standard artifacts dir, since test framework expects it

Reply:
I wouldn't expect it to interfere much tbh

Reply:
>For example, there're files that are generated during NuGet restore, and dependencies would be different between FSharp.sln and FSharp.Compiler.Service.sln

What files would be different?
Is this any different from regular project/solution structures where more than one solution contains a given project?
I'd hope that one day I can just run `dotnet restore src/compiler/FSharp.Compiler.Service.fsproj`. Is that the plan?